submit_async.go

```diff
@@ -0,0 +1,79 @@
+package gosnowflake
+
+// this is a sigma only file
+// when rebasing you can ignore everthing in this file as long as this
+// file makes it into master
+
+import (
+	"context"
+	"database/sql/driver"
+	"strconv"
+)
+
+type isComplete bool
+type queryID string
+
+func (sc *snowflakeConn) SubmitAsync(
+	ctx context.Context,
+	query string,
+	args []driver.NamedValue,
+) (queryID, isComplete, error) {
+	ctx = WithAsyncMode(WithAsyncModeNoFetch(ctx))
+	qid, err := getResumeQueryID(ctx)
+	if err != nil {
+		return "", false, err
+	}
+
+	// if query already submitted, return the query id
+	if qid != "" {
+		return queryID(qid), false, nil
```

Comment:
Should we still check if the query is complete?

Reply:
yeah thats a good idea